bundles/org.jkiss.utils/src/org/jkiss/utils/MutableBox.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.utils;
+
+/**
+ * A simple mutable container that holds only one value.
+ *
+ * <p>
+ * Among other situations, it's helpful when you need to generify a method
+ * with an effectively final variable without suppressing unchecked cast warning.
+ *
+ * @param <T> type of value
+ */
+public class MutableBox<T> {
```

Comment:
Why is this better than creating an array of a single element? Despite the fact that this class itself introduces an indirection, it will create additional overhead when using primitives by boxing them.

Additionally, it's only used in a single place. I don't think it worth it.

If you really want to use some kind of box type, then consider using `Reference<T>`

Reply:
You can use an `Object[1]` indeed, but it requires warning suppression in case of generic code like the newly introduced `syncExec(Supplier<T>)`. I'm not sure what you mean by indirection in this context (arrays themselves introduce indirection, don't they?). I'm also unsure what exact `Reference<T>` you mean.

Reply:
I'd prefer making this class to be private and nested if it's possible or avoid creating such utility class at all.

Reply:
OK then